src/metabase/api/app.clj

```diff
@@ -178,126 +179,233 @@
-                                                            "button.label" (i18n/tru "Edit"),
-                                                            "action_slug" "update"}}])}
+                {:name page-name
+                 :scaffold-target ["page" ident-type page-ident page-type]
+                 :ordered_cards (if (= "list" page-type)
+                                  (cond-> [{:size_y 12 :size_x 18 :row 1 :col 0
+                                            :card_id ["scaffold-target-id" "card" ident-type page-ident page-type]
+                                            :visualization_settings {"click_behavior"
+                                                                     {"type" "link"
+                                                                      "linkType" "page"
+                                                                      "parameterMapping" {(str "scaffold_" page-ident)
+                                                                                          {"source" {"type" "column",
+                                                                                                     "id" pk-field-name
+                                                                                                     "name" pk-field-name},
+                                                                                           "target" {"type" "parameter",
+                                                                                                     "id" (str "scaffold_" page-ident)},
+                                                                                           "id" (str "scaffold_" page-ident)}}
+                                                                      "targetId" ["scaffold-target-id" "page" ident-type page-ident "detail"]}}}]
+                                    (seq actions)
+                                    (concat (loop [[[slug action-name] & actions] actions
+                                                   col (- 16 (* 2 (count actions)))
+                                                   dashcards []]
+                                              (if slug
+                                                (recur
+                                                  actions
+                                                  (+ col 2)
+                                                  (conj dashcards
+                                                        {:size_y 1 :size_x 2 :row 0 :col col
+                                                         :card_id model-ref
+                                                         :visualization_settings {"virtual_card" {"display" "action"}
+                                                                                  "button.label" action-name
+                                                                                  "action_slug" slug}}))
```

Comment:
Wouldn't something like `(mapv (fn [[slug action-name] col] ...) actions (range (- 18 (* 2 (count actions))) Long/MAX_VALUE 2))` be more readable than the `loop`?